Code/Editor/Controls/ConsoleSCB.cpp

```diff
@@ -1080,6 +1079,33 @@ void ConsoleVariableModel::ClearModifiedRows()
     m_modifiedRows.clear();
 }
 
+AZ::ConsoleCommandInvokedEvent::Handler ConsoleVariableEditor::m_commandInvokedHandler(
+    [](AZStd::string_view command,
+       const AZ::ConsoleCommandContainer&,
+       AZ::ConsoleFunctorFlags,
+       AZ::ConsoleInvokedFrom)
+    {
+        // stop at the first space, if it exists, or npos if it doesn't
+        const auto spacePosition = command.find_first_of(" ");
+
+        AZStd::string variableName = command.substr(0, spacePosition);
+        if (variableName == AzToolsFramework::DocumentPropertyEditor::GetEnableDPECVarName())
+        {
+            // the cvar editor pref changed, unregister the old and register the new
+            AzToolsFramework::UnregisterViewPane(LyViewPane::ConsoleVariables);
+            ConsoleVariableEditor::RegisterViewClass();
+        }
+        else
+        {
+            // find the cvar that changed and keep the ConsoleVariableEditor informed
+            auto changedCVar = GetIEditor()->GetSystem()->GetIConsole()->GetCVar(variableName.c_str());
+            if (changedCVar)
+            {
+                OnVariableUpdated(changedCVar);
+            }
+        }
+    });
```

Comment:
This should be constructed in the ConsoleVariableEditor constructor as a member and shouldn't be static.

This isn't a singleton nor should it be.

Also as a recommendation, I think you should separate the initialization of the lambda to a separate variable.

```
```
```suggestion
    auto AzConsoleCommandInvoked = [this](AZStd::string_view command,
       const AZ::ConsoleCommandContainer&,
       AZ::ConsoleFunctorFlags,
       AZ::ConsoleInvokedFrom)
    {
        // stop at the first space, if it exists, or npos if it doesn't
        const auto spacePosition = command.find_first_of(" ");

        AZStd::string variableName = command.substr(0, spacePosition);
        if (variableName == AzToolsFramework::DocumentPropertyEditor::GetEnableDPECVarName())
        {
            // the cvar editor pref changed, unregister the old and register the new
            AzToolsFramework::UnregisterViewPane(LyViewPane::ConsoleVariables);
            RegisterViewClass();
        }
        else
        {
            // find the cvar that changed and keep the ConsoleVariableEditor informed
            auto changedCVar = GetIEditor()->GetSystem()->GetIConsole()->GetCVar(variableName.c_str());
            if (changedCVar)
            {
                OnVariableUpdated(changedCVar);
            }
        }
    };
    
    m_commandInvokedHandler = AZ::ConsoleCommandInvokedEvent::Handler(AZStd::move(AzConsoleCommandInvoked));
```

One reason to do this is that Visual Studio deals with formatting better in this case.
When a lambda is placed inside of a function call, the IDE adds an extra level of indent by the default.

Reply:
it's not a singleton, but it handles CVar changes when the ConsoleVariableEditor does not exist, and therefore must have a lifetime outside the scope of the class. If you look at ConsoleVariableEditor::RegisterViewClass, if the ShouldReplaceRPE CVar is set, it replaces the ConsoleVariableEditor as the registered viewpane and the ConsoleVariableEditor will not be constructed.

Reply:
Well this should be fine as long as non variables are bound to the lambda then.

Because this is static, the handler doesn't get destroyed until static de-initialization of the EditorLib module.